zubhub_frontend/zubhub/src/views/activities/activities.jsx

```diff
@@ -24,35 +24,40 @@ function Activities(props) {
   const classes = useStyles();
   const [loading, setLoading] = useState(true);
   const { activities } = useSelector(state => state);
-  let [activityList, setActivityList] = useState([]);
+  const [activityList, setActivityList] = useState(props?.activities ?? []);
 
   const commonClasses = makeStyles(DefaultStyles)();
 
-  useEffect(() => {
-    setActivityList(activities.all_activities);
-  }, [activities]);
+  const flagMap = useMemo(
+    () => ({
+      staff: () =>
+        props.getUnPublishedActivities({
+          t: props.t,
+          token: props.auth.token,
+        }),
+      educator: () =>
+        props.getMyActivities({
+          t: props.t,
+          token: props.auth.token,
+        }),
+    }),
+    [props],
+  );
 
-  const flagMap = {
-    staff: () =>
-      props.getUnPublishedActivities({
-        t: props.t,
-        token: props.auth.token,
-      }),
-    educator: () =>
-      props.getMyActivities({
-        t: props.t,
-        token: props.auth.token,
-      }),
-  };
-  useEffect(async () => {
-    setLoading(true);
-    if (location.state?.flag && flagMap[location.state.flag]) {
-      await flagMap[location.state.flag]();
-    } else {
-      await props.getActivities(props.t);
+  useEffect(() => {
+    async function fetcher() {
+      setLoading(true);
+      if (location.state?.flag && flagMap[location.state.flag]) {
+        await flagMap[location.state.flag]();
+      } else {
+        await props.getActivities(props.t);
+        await props.getMyActivities(props.auth);
```

Comment:
adding `myActivities` to `allActivities`  such that a user can see personally created routes on `/activities`.

Reply:
where are you proving the restriction to creators to this content?